### PR TITLE
WARN fixup warning in pandas/tests/apply/test_frame_apply.py

### LINKED ISSUE #50292: WARN fixup warning in pandas/tests/apply/test_frame_apply.py
We're not (yet) running all tests with `-W error`, so some warnings slip through to CI. On the way towards that, let's fixup this file.

Task here is:
1. make sure you've got your environment set up, see the [contributing guide](https://pandas.pydata.org/docs/dev/development/contributing.html)
2. make sure you can run this test: `pytest pandas/tests/apply/test_frame_apply.py`
3. fixup the warning (you just need to add `dayfirst=True` to the `to_datetime` call)
4. check that `pytest pandas/tests/apply/test_frame_apply.py` doesn't throw a warning
5. stage, commit, push, open pull request, celebrate 🎉 

Feel free to tag me for review

### PR DESCRIPTION
**Pull request by tpackard1**
- [ ] closes #50292 
- [ ] All code checks passed
@MarcoGorelli

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
you've got irrelevant commit here, can you do
```
git fetch upstream
git rebase -i upstream/main
```
then, in the pop-up, only select the last commit. then
```
git push origin HEAD --force
```

**Comment by tpackard1** [comment]
Thanks I was wondering why all the other commits were showing up.

File: doc/source/whatsnew/v0.19.0.rst [hunk 1]
```diff
@@ -654,7 +654,7 @@ Comparison operators raise ``ValueError`` when ``.index`` are different.
 
    In [3]: df1 == df2
    Out[3]:
-   ValueError: Can only compare identically-labeled DataFrame objects
+   ValueError: Can only compare identically-labeled (both index and columns) DataFrame objects
```
Comment (MarcoGorelli) [inline doc/source/whatsnew/v0.19.0.rst:657]: still got an irrelevant change here

**Review by MarcoGorelli — APPROVED** [review]
looks good to me pending green

**Pull request merged**